library/alloc/src/vec/mod.rs

```diff
@@ -1672,11 +1672,47 @@ impl<T, A: Allocator> Vec<T, A> {
         if self.len == self.buf.capacity() {
             self.reserve(1);
         }
+        unsafe { self.push_within_capacity(value).unwrap_unchecked() };
```

Comment:
Since this is probably hot code, I'd just leave this as it is since the unsafe code is trivial enough to duplicate (debug builds may have additional call/eq/unwrap overhead, which is undesirable imo)

Reply:
Yeah, let's see what perf says.

Reply:
It already breaks a UI test checking stack usage in debug mode 🤷‍♂️